src/sage/rings/semirings/tropical_variety.py

```diff
@@ -983,29 +1243,283 @@ def vertices(self):
+                    if (x,y) not in comp_vert:
+                        comp_vert[(x,y)] = [(i, 'neg')]
+                    else:
+                        comp_vert[(x,y)].append((i, 'neg'))
+        return comp_vert
+
+    def weight_vectors(self):
+        r"""
+        Return the weight vectors for all vertices of ``self``.
+
+        Weight vectors are a list of vectors associated with each vertex
+        of the curve. Each vector corresponds to an edge emanating from
+        that vertex and points in the direction of the edge.
+
+        Suppose `v` is a vertex adjacent to the edges `e_1, ldots, e_k`
+        with respective weights `w_1, \ldots, w_k`. Every edge `e_i` is
+        contained in a line (component) defined by an equation. Therefore,
+        there exists a unique integer vector `v_i = (\alpha, \beta)` in
+        the direction of `e_i` such that `\gcd(\alpha, \beta)=1`. Then,
+        each vertex `v` yield the vectors `w_1 v_1, ldots, w_k v_k`.
+        These vectors will satisfy the following balancing condition:
+        `\sum_{i=1}^k w_i v_i = 0`.
+
+        OUTPUT:
+
+        A dictionary where the keys represent the vertices, and the values
+        are lists of vectors.
+
+        EXAMPLES::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R.<x,y> = PolynomialRing(T)
+            sage: p1 = R(-2)*x^2 + R(-1)*x + R(1/2)*y + R(1/6)
+            sage: p1.tropical_variety().weight_vectors()
+            {(1, -1/2): [(0, 1), (-1, -2), (1, 1)],
+             (7/6, -1/3): [(-1, -1), (0, 1), (1, 0)]}
+
+            sage: p2 = R(2)*x^2 + x*y + R(2)*y^2 + x + R(-1)*y + R(3)
+            sage: p2.tropical_variety().weight_vectors()
+            {(-2, 0): [(-1, -1), (0, 1), (1, 0)],
+             (-1, -3): [(-1, -1), (0, 1), (1, 0)],
+             (-1, 0): [(-1, 0), (0, -1), (1, 1)],
+             (3, 4): [(-1, -1), (0, 1), (1, 0)]}
+        """
+        from sage.calculus.functional import diff
+        from sage.modules.free_module_element import vector
+        from sage.arith.misc import gcd
+
+        if not self._vertices_components():
+            return {}
+
+        # Calculate the base vector in the direction of each edge
+        temp_vectors = []
+        par = self._hypersurface[0][1][0].variables()[0]
+        for comp in self._hypersurface:
+            dx = diff(comp[0][0], par)
+            dy = diff(comp[0][1], par)
+            multiplier = gcd(QQ(dx), QQ(dy))
+            temp_vectors.append(vector([dx/multiplier, dy/multiplier]))
+
+        # Calculate the weight vectors of each vertex
+        cov = self._vertices_components()
+        result = {}
+        for vertex in cov:
+            vectors = []
+            for comp in cov[vertex]:
+                weight = self._hypersurface[comp[0]][2]
+                if comp[1] == 'pos':
+                    vectors.append(weight*temp_vectors[comp[0]])
+                else:
+                    vectors.append(weight*(-temp_vectors[comp[0]]))
```

Comment:
```suggestion
                vectors.append(weight*comp[1]*temp_vectors[comp[0]])
```
Plus having it return +/-1 means we can simplify this too.